tests/genai/scorers/phoenix/test_utils.py

```diff
@@ -0,0 +1,137 @@
+import sys
+from unittest.mock import Mock, patch
+
+import pytest
+
+from mlflow.exceptions import MlflowException
+
+
+def test_check_phoenix_installed_raises_without_phoenix():
+    with patch.dict("sys.modules", {"phoenix": None, "phoenix.evals": None}):
+        for mod in list(sys.modules.keys()):
+            if "mlflow.genai.scorers.phoenix" in mod:
+                del sys.modules[mod]
+
+        from mlflow.genai.scorers.phoenix.utils import check_phoenix_installed
+
+        with pytest.raises(MlflowException, match="arize-phoenix-evals"):
+            check_phoenix_installed()
+
+
+def test_map_scorer_inputs_basic():
+    from mlflow.genai.scorers.phoenix.utils import map_scorer_inputs_to_phoenix_record
+
+    record = map_scorer_inputs_to_phoenix_record(
+        inputs="What is MLflow?",
+        outputs="MLflow is a platform",
+    )
+
+    assert record["input"] == "What is MLflow?"
+    assert record["output"] == "MLflow is a platform"
+    assert "reference" not in record
+
+
+def test_map_scorer_inputs_with_expected_response():
+    from mlflow.genai.scorers.phoenix.utils import map_scorer_inputs_to_phoenix_record
+
+    record = map_scorer_inputs_to_phoenix_record(
+        inputs="What is MLflow?",
+        outputs="MLflow is a platform",
+        expectations={"expected_response": "MLflow is an ML platform."},
+    )
+
+    assert record["input"] == "What is MLflow?"
+    assert record["output"] == "MLflow is a platform"
+    assert record["reference"] == "MLflow is an ML platform."
+
+
+def test_map_scorer_inputs_with_context():
+    from mlflow.genai.scorers.phoenix.utils import map_scorer_inputs_to_phoenix_record
+
+    record = map_scorer_inputs_to_phoenix_record(
+        inputs="What is MLflow?",
+        outputs="MLflow is a platform",
+        expectations={"context": "MLflow context here."},
+    )
+
+    assert record["reference"] == "MLflow context here."
+
+
+def test_map_scorer_inputs_expected_response_priority():
+    from mlflow.genai.scorers.phoenix.utils import map_scorer_inputs_to_phoenix_record
+
+    record = map_scorer_inputs_to_phoenix_record(
+        inputs="test",
+        outputs="test output",
+        expectations={
+            "expected_response": "priority value",
+            "context": "should be ignored",
+            "reference": "also ignored",
+        },
+    )
+
+    assert record["reference"] == "priority value"
+
+
+def test_map_scorer_inputs_with_trace():
+    from mlflow.genai.scorers.phoenix.utils import map_scorer_inputs_to_phoenix_record
+
+    mock_trace = Mock()
+
+    with (
```

Comment:
is there a way we can avoid using mocks for this? It'd be great to construct a realistic trace and then test that the record is constructed properly as right now there's no real data being passed into the `map_scorer_inputs_to_phoenix_record` method. same for below.